Load components into confirm page using templates

jrichter1 opened the pull request:
This is a draft for templating confirm page, please take a look and criticise. (for the time being it even starts on confirm page, so it takes less time to debug)
- component panel template created
- some code/conditions moved from markup to the respective component
- generalized the error messages when some dependency is old (or is generally uninstallable)

To do (honestly could use some suggestions here):
- some mechanism for showing messages per component with arbitrary conditions (for now I've just left those in the confirm page)
- is there a better way to order the panels in this particular fashion, than reordering requirements.json?

dgolovin commented on browser/model/messages.json:
```diff
@@ -0,0 +1,27 @@
+{
```
I would say this information belongs to requirements.json file. It would be good to be able to update all info by downloading only one file from remote location (part of https://issues.jboss.org/browse/JBDS-4470).

dgolovin commented on browser/pages/confirm/confirm.html:
```diff
@@ -11,546 +11,22 @@
-          <span id="fuseplatform-name" class="product-name">{{checkboxModel.fuseplatform.productName}}</span><span id="fuseplatform-version" class="product-version">{{checkboxModel.fuseplatform.version}}</span>
-          <div id="fuseplatform-description">{{checkboxModel.fuseplatform.productDesc}}</div>
-        </div>
+      <div class="message-container error-message" ng-show="item.keyName === 'virtualbox' && !confCtrl.isDisabled && !checkboxModel.hyperv.isConfigured() && checkboxModel.virtualbox.virtualizationEnabled === false && (checkboxModel.cdk.selectedOption == 'install' || checkboxModel.virtualbox.selectedOption == 'install') && platform === 'win32'">
```
Feels like this 'simple messages' have:
1. Text
2. type (error/warning)
3. `function isVisible()` or `get isVisible()`

Then we would just iterate them in directive with ng-repeat.

jrichter1 commented:
I've moved the messages into requirements.json and also added some mechanism to evaluate arbitrary conditions from string. It's a bit raw, but it can be made prettier if needed, I believe.

dgolovin commented:
It looks good, except some white box unit and ui tests failed.

jrichter1 commented:
test this

jrichter1 commented:
test this

dgolovin commented:
fixed couple other issues in componentLoader code and PR is herehttps://github.com/redhat-developer-tooling/developer-platform-install/pull/808

jrichter1 commented:
should this one be closed then?

dgolovin commented:
Closing and pushing other one https://github.com/redhat-developer-tooling/developer-platform-install/pull/808 with additional fixes.
Thanks @jrichter1!
